src/NosCore.WorldServer/Controllers/StatController.cs

```diff
@@ -79,6 +80,13 @@ public IActionResult UpdateStats([FromBody] StatData data)
                     }
                     session.SetGold(data.Data);
                     break;
+                case UpdateStatActionType.UpdateClass:
+                    if ((byte) session.Class == data.Data)
```

Comment:
Maybe this check should be done in the ChangeClass method, this would seem more logical to me